src/calibration_pipeline/HelperFuncs.jl

```diff
@@ -36,3 +36,147 @@ function make_filter_props(n_elem, t_calib; apply = false, nz_per_filtered_cell
 
     return filter
 end
+
+function get_numbers_from_config(config::Dict)
+
+    n_cases = length(config["observations"]["cases"])
+    n_variables = length(config["observations"]["data_names"])
+    n_heights =
+        config["model"]["filter"]["apply"] ? config["model"]["filter"]["nz_filtered"] : config["model"]["n_elem"]
+    n_times =
+        config["model"]["filter"]["apply"] ? length(config["model"]["t_calib"]) - 1 : length(config["model"]["t_calib"])
+
+    return (n_cases = n_cases, n_variables = n_variables, n_heights = n_heights, n_times = n_times)
+end
+
+function make_block_diagonal_matrix(a::AbstractMatrix, b::AbstractMatrix)
+
+    if (a isa Diagonal) && (b isa Diagonal)
+        return Diagonal([diag(a); diag(b)])
+    end
+
+    (nr_a, nc_a) = size(a)
+    (nr_b, nc_b) = size(b)
+
+    nr_c = nr_a + nr_b
+    nc_c = nc_a + nc_b
+    c = zeros(nr_c, nc_c)
+    c[1:nr_a, 1:nc_a] = a
+    c[(nr_a + 1):nr_c, (nc_a + 1):nc_c] = b
+    return c
+end
+
+"""
+    lwp_rwp_rr(vec, config; isref)
+
+compute liquid water path [kg/m^2], rain water path [kg/m^2], and rain rate [mm/hr]
+
+# Inputs:
+- `vec` :: vector containing field information; 
+           must have the same structure as vectors in calibrations (height, variable, time, case)
+- `config` :: dictionary containing settings
+"""
+function lwp_rwp_rainrate(vec, config; isref = false)
```

Comment:
Maybe it would make sense to have some "diagnostics" folder and put the formulas for lwp, rwp and rain rate there.  In case we would want to to re-use them for plotting etc?

Reply:
If we do that we could make rain rate a function of density/height too, to allow for computing rain rate at any height, not just at the surface?